### PR TITLE
U10: The player can discard an item of his choice from his hand

### PR DESCRIPTION
**Pull request by paul891112**
I wasnt able to implement discarded locally, because to differentiate Drawn and Discarded state in ping data locally, the only way is to keep track of all discarded tiles locally. However, the record will get lost if I refresh the page.

To solve this problem, i make backend keep track of a boolean attribute "discarded" which evaluates to true if the tile is discarded. Also adjust code everywhere to match the change.

Might need to create a custom trash can icon

Let me know if you have any design suggestions (like where to place the trash icon or how it should look like)

### CODE REVIEW

**Comment by Croxsy** [comment]
> I could do option 1 because its easier to implement
> 
> Do you mean place it "over" the bottom right corner of tile?
> 
> And no, sorry i couldnt find any icon without background, all the images are downloaded in square shape, all i can do is minimize the white background.

Yeah I mean over:) Okay so we'll do an trash icon ourselfes

**Comment by paul891112** [comment]
Set tile state to Discarded is now done in backend, frontend just makes the call and receives tile information from ping

File: src/scenes/gameuiscreen.ts [hunk 1]
```diff
@@ -134,6 +135,40 @@ export class GameUiScreen extends Phaser.Scene {
                 new Phaser.Math.Vector2(tile.x, tile.y),
             );
             this.drawnTilesContainer.add(tile);
+
+            //Add trash can icon over every drawn tile
+            //x = tileCoordinateX + tilePixels/2 - trashcanPixels/2
+            //y = tileCoordinateY + tilePixels/2 - trashcanPixels/2
```
Comment (C0DECYCLE) [inline src/scenes/gameuiscreen.ts:141]: i think does comments are not necessary

File: src/scenes/gameuiscreen.ts [hunk 2]
```diff
@@ -134,6 +135,40 @@ export class GameUiScreen extends Phaser.Scene {
...
+                "pointerdown",
+                () => {
+                    trashCan.destroy();
+                    this.dragObj = tile;
+                    this.currentTile.id = myTiles[i].id;
+                    this.currentTile.type = myTiles[i].type;
+                    this.discardTile();
+                },
+                this,
+            );
```
Comment (C0DECYCLE) [inline src/scenes/gameuiscreen.ts:169]: we could use the built in interactify utils function here

**Review by C0DECYCLE — APPROVED** [review]
Also sorry here for the delay. I did test everything, works super. Code looks also very good only 2 minor things inside the uiscreen.

**Review by C0DECYCLE — CHANGES_REQUESTED** [review]
Sorry accidentally selected the wrong field.

**Pull request merged**